sdk/communication/chat/src/main/java/com/azure/android/communication/chat/implementation/AzureCommunicationChatServiceImpl.java

```diff
@@ -0,0 +1,2309 @@
+
+        @GET("/chat/threads")
+        Call<ResponseBody> listChatThreads(@Query("maxPageSize") Integer maxPageSize, @Query("startTime") OffsetDateTime startTime, @Query("api-version") String apiVersion);
+
+        @PATCH("/chat/threads/{chatThreadId}")
+        Call<ResponseBody> updateChatThread(@Path("chatThreadId") String chatThreadId, @Query("api-version") String apiVersion, @Body RequestBody body);
+
+        @GET("/chat/threads/{chatThreadId}")
+        Call<ResponseBody> getChatThread(@Path("chatThreadId") String chatThreadId, @Query("api-version") String apiVersion);
+
+        @DELETE("/chat/threads/{chatThreadId}")
+        Call<ResponseBody> deleteChatThread(@Path("chatThreadId") String chatThreadId, @Query("api-version") String apiVersion);
+
+        @GET("{nextLink}")
+        Call<ResponseBody> listChatReadReceiptsNext(@Path(value = "nextLink", encoded = true) String nextLink);
+
+        @GET("{nextLink}")
+        Call<ResponseBody> listChatMessagesNext(@Path(value = "nextLink", encoded = true) String nextLink);
+
+        @GET("{nextLink}")
+        Call<ResponseBody> listChatThreadMembersNext(@Path(value = "nextLink", encoded = true) String nextLink);
+
+        @GET("{nextLink}")
+        Call<ResponseBody> listChatThreadsNext(@Path(value = "nextLink", encoded = true) String nextLink);
+    }
+
+    /**
+     * Gets read receipts for a thread.
+     * 
+     * @param chatThreadId Thread id to get the read receipts for.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatReadReceipts(String chatThreadId, final Callback<Page<ReadReceipt>> callback) {
+        Call<ResponseBody> call = service.listChatReadReceipts(chatThreadId, this.getApiVersion());
+        retrofit2.Callback<ResponseBody> retrofitCallback = new retrofit2.Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<okhttp3.ResponseBody> call, retrofit2.Response<ResponseBody> response) {
+                if (response.isSuccessful()) {
+                    if (response.code() == 200) {
+                        final ReadReceiptsCollection decodedResult;
+                        try {
+                            decodedResult = deserializeContent(response.headers(), response.body(), ReadReceiptsCollection.class);
+                        } catch(Exception ex) {
+                            callback.onFailure(ex, response.raw());
+                            return;
+                        }
+                        callback.onSuccess(new Page<ReadReceipt>(response.raw().request().url().toString(), decodedResult.getValue(), decodedResult.getNextLink()), response.raw());
```

Comment:
How sure are we that some of these calls won't throw a `NullPointerException` at some point?

Reply:
it shouldn't, provided we arrived at this point means, we manage to make a request to an endpoint i.e. to a valid URI.

Reply:
Good point. We need to address that in codegen